### ISSUE TITLE
[ARM] policy definition: example using parameters

### ISSUE THREAD

**Issue by jtuliani** [Azure/azure-cli#5810]
Needs an example showing how to create a policy definition that uses parameters.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 436bf02d-dfe3-ef6e-2be8-1bd7b1f39623
* Version Independent ID: e8b92df3-9e3d-c4c7-1965-0b6b3f7d00ea
* Content: [az policy definition | Microsoft Docs](https://docs.microsoft.com/en-us/cli/azure/policy/definition?view=azure-cli-latest)
* Content Source: [src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_help.py](https://github.com/Azure/azure-cli/blob/master/src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_help.py)
* Service: **azure-resource-manager**
* GitHub Login: @rloutlaw
* Microsoft Alias: **routlaw**

**Comment by kiazhi** [comment]
Discovered that the "Create a policy parameter definition with the following example" is an incorrect example. Parameter JSON cannot be passed to `--rules` parameter.

```sh
az policy definition create -n allowedLocations --rules '{
                            "allowedLocations": {
                                "type": "array",
                                "metadata": {
                                    "description": "The list of locations that can be specified when deploying resources",
                                    "strongType": "location",
                                    "displayName": "Allowed locations"
                                }
                            }
                        }'
```

**Comment by yugangw-msft** [comment]
Fixed by #7470
